src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx

```diff
@@ -116,21 +116,28 @@ function MoneyRequestPreviewContent({
     const isFetchingWaypointsFromServer = TransactionUtils.isFetchingWaypointsFromServer(transaction);
     const isCardTransaction = TransactionUtils.isCardTransaction(transaction);
     const isSettled = ReportUtils.isSettled(iouReport?.reportID);
+    const isApproved = ReportUtils.isReportApproved(iouReport);
     const isDeleted = action?.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
     const isReviewDuplicateTransactionPage = route.name === SCREENS.TRANSACTION_DUPLICATE.REVIEW;
 
     const isFullySettled = isSettled && !isSettlementOrApprovalPartial;
-    const isFullyApproved = ReportUtils.isReportApproved(iouReport) && !isSettlementOrApprovalPartial;
+    const isFullyApproved = isApproved && !isSettlementOrApprovalPartial;
 
     // Get transaction violations for given transaction id from onyx, find duplicated transactions violations and get duplicates
-    const duplicates = useMemo(
+    const allDuplicates = useMemo(
         () =>
             transactionViolations?.[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transaction?.transactionID}`]?.find(
                 (violation) => violation.name === CONST.VIOLATIONS.DUPLICATED_TRANSACTION,
             )?.data?.duplicates ?? [],
         [transaction?.transactionID, transactionViolations],
     );
 
+    // Remove settled transactions from duplicates
+    const duplicates = useMemo(() => TransactionUtils.removeSettledAndApprovedTransactions(allDuplicates), [allDuplicates]);
+
+    // When there are no settled transactions in duplicates, show the "Keep this one" button
+    const shouldShowKeepButton = allDuplicates.length === duplicates.length;
```

Comment:
We need to check if there are any elements in `duplicates` before making this comparison. More details here #48366